internal/api/courses/admin/email.go

```diff
@@ -0,0 +1,66 @@
+package admin
+
+import (
+	"errors"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/email"
+)
+
+type EmailRequest struct {
+	core.APIRequestCourseUserContext
+	core.MinCourseRoleGrader
+
+	email.Message
+
+	DryRun bool `json:"dry-run"`
+}
+
+type EmailResponse struct {
+	To  []string `json:"to"`
+	CC  []string `json:"cc"`
+	BCC []string `json:"bcc"`
+}
+
+// Send an email to course users.
+func HandleEmail(request *EmailRequest) (*EmailResponse, *core.APIError) {
+	response := EmailResponse{}
```

Comment:
Since you don't use this until then end, I would just define it as a literal and set To, CC, and BCC at the same time.
(Don't do it in the return, that looks messy with multiple returns.)

Generally, we want to declare variable right before they are used.